auth/env_keychain.go

```diff
@@ -12,15 +12,30 @@ import (
 	"github.com/pkg/errors"
 )
 
-// NewKeychain returns either a EnvKeychain or a authn.DefaultKeychain depending on whether the provided environment variable is set
-func NewKeychain(envVar string) authn.Keychain {
+// NewKeychain returns an EnvKeychain based on the provided environment variable.
+// If the provided environment variable is not set, NewKeychain will build an auth config
+// from authn.DefaultKeychain and set the variable in the environment.
+// If any errors are encountered, NewKeychain will return an anonymous keychain.
+func NewKeychain(envVar string, images ...string) authn.Keychain {
```

Comment:
We are wondering if `NewKeychain` is still a good name for this function. It seems like `GetKeychain` might be better. We are assuming that we can't change it, since it's exported, but maybe we're wrong.

Reply:
On second thought, maybe it's better if we don't modify the behavior of `NewKeychain` and create a separate function (say, `SetEnvVar`) that we can call before dropping privileges.

Reply:
We went ahead and made this change (`SetEnvVar`), it seemed like the right thing to do and allowed us to delete some code.